pkg/ko/publish/options.go

```diff
@@ -42,21 +43,24 @@ func WithAuth(auth authn.Authenticator) Option {
 // authenticator on a default publisher using an authn.Keychain
 func WithAuthFromKeychain(keys authn.Keychain) Option {
 	return func(i *defaultOpener) error {
-		auth, err := keys.Resolve(i.base.Registry)
+		repo, err := name.NewRepository(i.base, name.WeakValidation)
```

Comment:
Should we also make this change in [`pkg/v1/remote/options.go`](https://github.com/google/go-containerregistry/blob/master/pkg/v1/remote/options.go#L41)?

Reply:
No, that still takes a parsed image name.  We can't here because we parse DockerHub base references of the form: `docker.io/mattmoor` as `docker.io/library/mattmoor` :(

Reply:
Can you add a comment to that effect? I'm trying to keep some idiot (me) from coming by in a few months and trying to collapse both `WithAuthFromKeychain` implementations into one, and causing pain.

Reply:
Done